js/base/Exchange.js

```diff
@@ -1634,6 +1634,12 @@ module.exports = class Exchange {
         return market['symbol'];
     }
 
+    safeMarketSymbol (o, k, market = undefined, delimiter = undefined) {
+        const marketId = this.safeString (o, k)
+        market = this.safeMarket (marketId, market, delimiter)
```

Comment:
Let's avoid single-letter variable names for all variables except counters. Also, the call at line 1639 is not necessary, since it's done inside `safeSymbol()`:

```suggestion
    safeMarketSymbol (dictionary, key, market = undefined, delimiter = undefined) {
        const marketId = this.safeString (dictionary, key)
```

Reply:
yeah, the repeated `safe` methods were mistake, i was away for a moment and missed copy-pasted line.
about dictionary keys - thnx, right, i've changed and will note for future.

Reply:
also, i will only submit JS method for the last PR's and when transpiler is ready, then can be merged so they will be transpiled into other langs too.